composer.json

```diff
@@ -40,9 +40,10 @@
         "php-cs-fixer/accessible-object": "^1.0",
         "php-cs-fixer/phpunit-constraint-isidenticalstring": "^1.2",
         "php-cs-fixer/phpunit-constraint-xmlmatchesxsd": "^1.2.1",
-        "phpunit/phpunit": "^5.7.27 || ^6.5.14 || ^7.1",
+        "phpunit/phpunit": "^5.7.27 || ^6.5.14 || ^7.5.20 || ^8.5.13 || ^9.4.4 <9.5",
         "phpunitgoodpractices/polyfill": "^1.5",
         "phpunitgoodpractices/traits": "^1.9.1",
+        "sanmai/phpunit-legacy-adapter": "^6 || ^8",
```

Comment:
You could use `Symfony\Bridge\PhpUnit\SetUpTearDownTrait` instead of adding another dependency that solves the same problem.

Reply:
Yup, `SetUpTearDownTrait` _theoretically_ solves the same problem. I tried to switch (it's just several `sed` commands) and I couldn't make it work for some reason. And I no longer have the luxury of time to dig deeper.

Therefore, I think the best course of action will be:
- We get this working/merged as it is. This PR has a lot more of things changed other that setUp/tearDown.
- Someone with excess time comes and runs `sed -i s/legacySetUp/goSetUp/g` etc, finds the root of the problem I had with these, and opens a PR to remove the extra dependency.

Reply:
this is the reason : https://github.com/symfony/symfony/pull/35311

i don't want to import the trait to all 400 test files, let's keep the current proposal

Reply:
Yup, that explains the failures I've been seeing.